packages/medusa/src/types/common.ts

```diff
@@ -14,36 +14,88 @@ export interface FindConfig<Entity> {
 
 export type PaginatedResponse = { limit: number; offset: number; count: number }
 
+export class DateComparisonOperator {
+  @IsDate()
+  lt?: Date | string
+
+  @IsDate()
+  gt?: Date | string
+
+  @IsDate()
+  gte?: Date | string
+
+  @IsDate()
+  lte?: Date | string
+}
```

Comment:
Im currently looking at getting this in a pr later today, so lets get this merged 😄